src/Sylius/Bundle/GridBundle/Doctrine/PHPCRODM/ExpressionBuilder.php

```diff
@@ -0,0 +1,171 @@
+
+/**
+ * ExpressionBuilder which builds Doctrine Commons
+ * ExpressionBuilder object graph which we can then walk in order
+ * to build up the PHPCR-ODM query builder.
+ */
+class ExpressionBuilder implements ExpressionBuilderInterface
+{
+    private $expressionBuilder;
+    private $orderBys = [];
+
+    function __construct()
+    {
+        $this->expressionBuilder = new CollectionsExpressionBuilder();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function andX($expressions)
+    {
+        return $this->expressionBuilder->andX($expressions);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function orX($expressions)
+    {
+        return $this->expressionBuilder->orX($expressions);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function comparison($field, $operator, $value)
+    {
+        throw new \BadMethodCallException('Not supported yet.');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function equals($field, $value)
+    {
+        return $this->expressionBuilder->eq($field, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function notEquals($field, $value)
+    {
+        return $this->expressionBuilder->neq($field, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lessThan($field, $value)
+    {
+        return $this->expressionBuilder->lt($field, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lessThanOrEqual($field, $value)
+    {
+        return $this->expressionBuilder->lte($field, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function greaterThan($field, $value)
+    {
+        return $this->expressionBuilder->gt($field, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function greaterThanOrEqual($field, $value)
+    {
+        return $this->expressionBuilder->gte($field, $value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function in($field, array $values)
+    {
+        return $this->expressionBuilder->in($field, $values);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function notIn($field, array $values)
+    {
+        return $this->expressionBuilder->notIn($field, $values);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isNull($field)
+    {
+        // could probably do this by comparing with eq = NULL and handling this case
+        // in the visitor.
+        throw new \BadMethodCallException('Not currently supported.');
+        return $this->expressionBuilder->notIn($field);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isNotNull($field)
+    {
+        throw new \BadMethodCallException('Not currently supported.');
+        return $this->expressionBuilder->isNull($field);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function like($field, $pattern)
+    {
+        return $this->expressionBuilder->contains($field, $pattern);
```

Comment:
`like` is analasgous to contains if I understand correctly (it is referenced to as "contains" in the filter).